diff --git a/custom_models/SI/si.world b/custom_models/SI/si.world
index b4cf36a..e18183b 100644
--- a/custom_models/SI/si.world
+++ b/custom_models/SI/si.world
@@ -237,36 +237,6 @@
         <plugin filename="libradiation_source_plugin.so" name="radiation_source_plugin" />
       </sensor>
     </link>
-  </model>
-  <model name="sensor_0">
-    <pose>1 2 0 0 0 0</pose> <!-- x y z roll pitch yaw -->
-    <static>true</static>
-    <link name="link">
-      <collision name="collision">
-        <geometry>
-          <box>
-            <size>0.1 0.1 0.1</size> <!-- length width height -->
-          </box>
-        </geometry>
-      </collision>
-      <visual name="visual">
-        <geometry>
-          <box>
-            <size>0.1 0.1 0.1</size>
-          </box>
-        </geometry>
-        <material>
-          <ambient>1 0 0 1</ambient> <!-- RGBA for green -->
-          <diffuse>1 0 0 1</diffuse>
-        </material>
-      </visual>
-      <sensor name="radiation_sensor_0" type="radiation_sensor">
-        <always_on> 1 </always_on>
-        <update_rate> 1.0 </update_rate>
-        <topic>source_0</topic>
-        <plugin filename="libradiation_sensor_plugin.so" name="radiation_sensor_plugin" />
-      </sensor>
-    </link>
   </model>
     <gui fullscreen='0'>
       <camera name='user_camera'>
diff --git a/launch/si.launch b/launch/si.launch
index 0d4722b..140733d 100644
--- a/launch/si.launch
+++ b/launch/si.launch
@@ -23,4 +23,8 @@
   <node name="gazebo" pkg="gazebo_radiation_plugins" type="gazebo"  output="screen"
   args="--verbose $(arg world_name)" />
 
+
+ <!-- Spawn sensor -->
+ <node name="spawn_mysensor_model" pkg="gazebo_ros" type="spawn_model" output="screen" 
+  args="-file $(find gazebo_radiation_plugins)/custom_models/SI/models/mysensor.sdf -sdf -model sensor_0" />
 </launch>
